feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SessionDetailFragment.kt

```diff
@@ -137,194 +138,63 @@ class SessionDetailFragment : DaggerFragment() {
-            textView.text = fullDescription
-            //Return here if not more than the specified number of rows
-            if (!(textView.lineCount > ELLIPSIS_LINE_COUNT && showEllipsis)) return@doOnPreDraw
-            val lastLineStartPosition = textView.layout.getLineStart(ELLIPSIS_LINE_COUNT - 1)
-            val ellipsis = getString(R.string.ellipsis_label)
-            val lastLineText = TextUtils.ellipsize(
-                fullDescription.substring(lastLineStartPosition),
-                textView.paint,
-                textView.width - textView.paint.measureText(ellipsis),
-                TextUtils.TruncateAt.END
-            )
-            val ellipsisColor =
-                ContextCompat.getColor(requireContext(), R.color.design_default_color_secondary)
-            val onClickListener = {
-                TransitionManager.beginDelayedTransition(binding.sessionLayout)
-                textView.text = fullDescription
-                showEllipsis = !showEllipsis
-            }
-            val detailText = fullDescription.substring(0, lastLineStartPosition) + lastLineText
-            val text = buildSpannedString {
-                clickableSpan(onClickListener, {
-                    append(detailText)
-                    color(ellipsisColor) {
-                        append(ellipsis)
-                    }
-                })
-            }
-            textView.setText(text, TextView.BufferType.SPANNABLE)
-            textView.movementMethod = LinkMovementMethod.getInstance()
-        }
-    }
-
-    private fun SpannableStringBuilder.clickableSpan(
-        clickListener: () -> Unit,
-        builderAction: SpannableStringBuilder.() -> Unit
-    ) {
-        inSpans(object : ClickableSpan() {
-            override fun onClick(widget: View) {
-                clickListener()
-            }
-
-            override fun updateDrawState(ds: TextPaint) {
-                // nothing
-            }
-        }, builderAction)
-    }
-
-    private fun ViewGroup.bindSpeaker(session: Session) {
-        removeAllViews()
-        (session as? SpeechSession)?.speakers.orEmpty().indices.forEach { index ->
-            val speaker: Speaker =
-                (session as? SpeechSession)?.speakers?.getOrNull(index) ?: return@forEach
-            val speakerView = layoutInflater.inflate(
-                R.layout.layout_speaker_session_detail, this, false
-            ) as ViewGroup
-            val speakerNameView = speakerView.findViewById<TextView>(R.id.speaker)
-            val speakerImageView = speakerView.findViewById<ImageView>(R.id.speaker_image)
-            speakerImageView.transitionName = "${speaker.id}-${TRANSITION_NAME_SUFFIX}"
-            speakerView.setOnClickListener {
-                val extras = FragmentNavigatorExtras(
-                    speakerImageView to speakerImageView.transitionName
+        if (binding.sessionDetailRecycler.adapter == null) {
+            val adapter = GroupAdapter<ViewHolder<*>>()
```

Comment:
In order to enable DiffUtil of RecyclerView, this adapter is defined by using "by autoCleared ()" for the property and assigned in onViewCreated. 🙏